pkg/backend/snapshot.go

```diff
@@ -249,18 +242,30 @@ func (ssm *sameSnapshotMutation) End(step deploy.Step, successful bool) error {
 	contract.Assert(successful)
 	logging.V(9).Infof("SnapshotManager: sameSnapshotMutation.End(..., %v)", successful)
 	return ssm.manager.mutate(func() bool {
+		sameStep := step.(*deploy.SameStep)
+
 		ssm.manager.markDone(step.Old())
+
+		// In the case of a 'resource create' in a program that wasn't specified by the user in the
+		// --target list, we *never* want to write this to the checkpoint.  We treat it as if it
+		// doesn't exist at all.  That way when the program runs the next time, we'll actually
+		// create it.
+		if sameStep.IsSkippedCreate() {
```

Comment:
```suggestion
		if sameStep := step.SameStep(); sameStep.IsSkippedCreate() {
```

And remote the assignment at the beginning of the function.

Reply:
so this only makes sameStep visible in the body of the 'if'.  but we want to reference it below as we pass it to `.mustWrite`.

Reply:
Ah, missed that bit. 👍